bigtop-tests/smoke-tests/hive/HiveJdbcGeneralTest.java

```diff
@@ -0,0 +1,390 @@
+            + "('Peter''s Auto Parts', 65.79, 7288, 'Apex'),"
+            + "('Deluxe Hardware', 467.12, 8941, 'Garner'),"
+            + "('Pauline''s Antiques', 302.05, 9112, 'Morrisville'),"
+            + "('Apex Catering', 37.95, 9923, 'Apex')";
+
+    String sortQuery =
+        "WITH `subquery0` AS ( SELECT `accountnumber` AS `accountnumber`, `company` AS `company`, `debt` AS `debt`, `town` AS `town` FROM btests ) SELECT `table0`.`company`, `table0`.`debt`, `table0`.`accountnumber`, `table0`.`town` FROM ( SELECT `accountnumber`, `company`, `debt`, `town` FROM ( SELECT `accountnumber`, `company`, `debt`, `town`, ROW_NUMBER() OVER ( PARTITION BY `town` ORDER BY CASE WHEN `town` IS NULL THEN 0 ELSE 1 END, `town` ) AS `tempcol0` FROM `subquery0` ) AS `subquery1` WHERE ( `tempcol0` = 1 ) ) AS `table0` ORDER BY CASE WHEN `table0`.`town` IS NULL THEN 0 ELSE 1 END, `table0`.`town`";
+
+    try (Statement stmt = con.createStatement()) {
+      dropTable(stmt, newTableName + "S");
+      stmt.executeUpdate(
+          "create table btests (company varchar(22), debt double, accountnumber double, town varchar(13))");
+      stmt.executeUpdate(queryValues);
+      assertEquals("Paulines Antiques", printResults(stmt, sortQuery));
+      dropTable(stmt, newTableName + "S");
+    }
+  }
+
+  @Test
+  public void testSpecifiedLocation() throws Exception {
+
+    String queryValues = "insert into table `class` values"
+        + "('Alfred', 'M', 14, 69, 112.5)," + "('Alice', 'F', 13, 56.5, 84),"
+        + "('Barbara', 'F', 13, 65.3, 98)," + "('Carol', 'F', 14, 62.8, 102.5),"
+        + "('Henry', 'M', 14, 63.5, 102.5)," + "('James', 'M', 12, 57.3, 83),"
+        + "('Jane', 'F', 12, 59.8, 84.5)," + "('Janet', 'F', 15, 62.5, 112.5),"
+        + "('Jeffrey', 'M', 13, 62.5, 84)," + "('John', 'M', 12, 59, 99.5),"
+        + "('Joyce', 'F', 11, 51.3, 50.5)," + "('Judy', 'F', 14, 64.3, 90),"
+        + "('Louise', 'F', 12, 56.3, 77)," + "('Mary', 'F', 15, 66.5, 112),"
+        + "('Philip', 'M', 16, 72, 150)," + "('Robert', 'M', 12, 64.8, 128),"
+        + "('Ronald', 'M', 15, 67, 133)," + "('Thomas', 'M', 11, 57.5, 85),"
+        + "('William', 'M', 15, 66.5, 112)";
+
+    String locationQuery = "CREATE TABLE test42 ROW FORMAT SERDE"
+        + "'org.apache.hadoop.hive.serde2.columnar.ColumnarSerDe' STORED AS RCFILE"
+        + " LOCATION '/tmp/test1' AS SELECT  `CLASS`.`age`,"
+        + " `CLASS`.`name`, `CLASS`.`sex`, `CLASS`.`height`, `CLASS`.`weight`  FROM"
+        + " `CLASS`";
+
+    try (Statement stmt = con.createStatement()) {
+      dropTable(stmt, "class");
+      dropTable(stmt, "test42");
+      stmt.executeUpdate(
+          "create table `class` (name varchar(8), sex varchar(1), age double precision, height double precision, weight double precision)");
+      stmt.executeUpdate(queryValues);
+      // specifying the location fails in hortonworks 3.0
+      try {
+        stmt.execute(locationQuery);
+      } catch (Exception e) {
+

```

Comment:
Not quite sure what you'd like to test here. There's no assertion in this test.
Perhaps asserting an exception?

Reply:
Yep, the location should be invalid as a new directory has been created. Asserting that the location is invalid.

Reply:
If no exception is thrown then there is no way to know which directory the file was written to (given that transactions now result in a more complex and verbose file strucutre)

Reply:
Hey @bharatm363  sorry I didn't get your point. Probably because I didn't ask clearly.
Let me rephrase my question. For this test case: testSpecifiedLocation, you'd like to test a statement that creates table with location specified. So the result is either:
a) the statement success
b) the statement failed
Now, we need to know whether the result is a or b. But here you catch the exception with no-op. So to my knowledge either a or b this test is marked success. Am I correct?